src/renderer/util/index.ts

```diff
@@ -41,26 +71,57 @@ export async function determineFilesFromNestedPaths(
 ): Promise<string[]> {
   const filePaths = await Promise.all(
     paths.flatMap(async (fullPath) => {
-      const stats = await fsPromises.stat(fullPath);
-      if (!stats.isDirectory()) {
-        return [fullPath];
-      }
-      const canRead = await canUserRead(fullPath);
-      if (!canRead) {
-        throw new Error(`User does not have permission to read ${fullPath}`);
-      }
-      const pathsUnderFolder = await fsPromises.readdir(fullPath, {
-        withFileTypes: true,
-      });
-      return pathsUnderFolder
-        .filter((f) => f.isFile())
-        .map((f) => resolve(fullPath, f.name));
+      return determineFilesFromNestedPath(fullPath);
     })
   );
 
   return uniq(flatten(filePaths));
 }
 
+/**
+ * Use a given filepath's extension to determine if it is a "multifile".
+ * @param filePath Path to the file
+ */
+export function determineIsMultifile(filePath: string): boolean {
+  const multifileExtensions = ['.zarr', '.sldy'];
+  const combinedExtensions = multifileExtensions.join('|');
+
+  // "ends with one of the listed extensions, ignoring casing"
+  // otherwise written like: /.zarr|.sldy$/i
+  const matcher = new RegExp(combinedExtensions + "$", 'i');
+
+  // If the regex matches it will return an array (truthy).
+  // If the regex doesn't match it will return null (falsy).
+  return Boolean(filePath.match(matcher)) && USE_MULTIFILE_ASSUMPTION;
+}
+
+/**
+ * Returns the total size of a given directory's children, sub-children, etc.
```

Comment:
What happens when following links?  possibly circular links?  Not a blocker, but definitely something to think about.

Reply:
Oh my. Links? Like symlinks? That's definitely not a use-case I was considering... Does FSS already account for that in some way on its end?

Reply:
Never thought of it either.  Worth testing, maybe blocking if it causes any problems.

Reply:
Fss stores the subfiles in a list, and checks that entries aren't already in the list before adding them.

Reply:
Does this links question seem worth an issue?

Reply:
Maybe a TODO comment in the code?  It shouldn't be blocker, though.